### PR TITLE
[FLINK-7408] [conf] Create WebOptions for WebRuntimeMonitor

### PR DESCRIPTION
**Pull request by tillrohrmann**
## What is the purpose of the change

This commit moves the WebRuntimeMonitor related configuration options from
JobManagerOptions to WebOptions. Moreover, it removes the prefix jobmanager.

As you've requested @zentol.

## Verifying this change

This change is a trivial rework / code cleanup without any test coverage.

## Does this pull request potentially affect one of the following parts:

  - Dependencies (does it add or upgrade a dependency): (no)
  - The public API, i.e., is any changed class annotated with `@Public(Evolving)`: (yes)
  - The serializers: (no)
  - The runtime per-record code paths (performance sensitive): (no)
  - Anything that affects deployment or recovery: JobManager (and its components), Checkpointing, Yarn/Mesos, ZooKeeper: (no)

## Documentation

  - Does this pull request introduce a new feature? (no)

### CODE REVIEW

File: flink-runtime/src/main/scala/org/apache/flink/runtime/minicluster/FlinkMiniCluster.scala [hunk 1]
```diff
@@ -388,7 +388,7 @@ abstract class FlinkMiniCluster(
     : Option[WebMonitor] = {
     if(
       config.getBoolean(ConfigConstants.LOCAL_START_WEBSERVER, false) &&
-        config.getInteger(JobManagerOptions.WEB_PORT.key(), 0) >= 0) {
+        config.getInteger(WebOptions.PORT.key(), 0) >= 0) {
```
Comment (zentol) [inline flink-runtime/src/main/scala/org/apache/flink/runtime/minicluster/FlinkMiniCluster.scala]: you can drop the call to `key()`.

File: flink-yarn/src/main/java/org/apache/flink/yarn/YarnApplicationMasterRunner.java [hunk 2]
```diff
@@ -510,8 +511,8 @@ private static Configuration createConfiguration(String baseDirectory, Map<Strin
 		}
 
 		// if a web monitor shall be started, set the port to random binding
-		if (configuration.getInteger(JobManagerOptions.WEB_PORT.key(), 0) >= 0) {
-			configuration.setInteger(JobManagerOptions.WEB_PORT, 0);
+		if (configuration.getInteger(WebOptions.PORT.key(), 0) >= 0) {
```
Comment (zentol) [inline flink-yarn/src/main/java/org/apache/flink/yarn/YarnApplicationMasterRunner.java]: drop call to `key()`

File: flink-yarn/src/main/java/org/apache/flink/yarn/entrypoint/YarnEntrypointUtils.java [hunk 3]
```diff
@@ -104,8 +105,8 @@ public static Configuration loadConfiguration(String workingDirectory, Map<Strin
 		}
 
 		// if a web monitor shall be started, set the port to random binding
-		if (configuration.getInteger(JobManagerOptions.WEB_PORT.key(), 0) >= 0) {
-			configuration.setInteger(JobManagerOptions.WEB_PORT, 0);
+		if (configuration.getInteger(WebOptions.PORT.key(), 0) >= 0) {
```
Comment (zentol) [inline flink-yarn/src/main/java/org/apache/flink/yarn/entrypoint/YarnEntrypointUtils.java]: drop call to `key()`

File: flink-runtime/src/main/scala/org/apache/flink/runtime/jobmanager/JobManager.scala [hunk 4]
```diff
@@ -2216,7 +2216,7 @@ object JobManager {
     : (ActorRef, ActorRef, Option[WebMonitor], Option[ActorRef]) = {
 
     val webMonitor: Option[WebMonitor] =
-      if (configuration.getInteger(JobManagerOptions.WEB_PORT.key(), 0) >= 0) {
+      if (configuration.getInteger(WebOptions.PORT.key(), 0) >= 0) {
```
Comment (zentol) [inline flink-runtime/src/main/scala/org/apache/flink/runtime/jobmanager/JobManager.scala]: drop call to `key()`

File: flink-runtime/src/main/java/org/apache/flink/runtime/clusterframework/BootstrapTools.java [hunk 5]
```diff
@@ -188,7 +189,7 @@ public static WebMonitor startWebMonitorIfConfigured(
 		config.setString(JobManagerOptions.ADDRESS, address.host().get());
 		config.setInteger(JobManagerOptions.PORT, Integer.parseInt(address.port().get().toString()));
 
-		if (config.getInteger(JobManagerOptions.WEB_PORT.key(), 0) >= 0) {
+		if (config.getInteger(WebOptions.PORT.key(), 0) >= 0) {
```
Comment (zentol) [inline flink-runtime/src/main/java/org/apache/flink/runtime/clusterframework/BootstrapTools.java]: drop call to `key()`

File: flink-runtime-web/src/main/java/org/apache/flink/runtime/webmonitor/WebRuntimeMonitor.java [hunk 6]
```diff
@@ -214,7 +214,7 @@ public WebRuntimeMonitor(
 		ExecutionContextExecutor context = ExecutionContext$.MODULE$.fromExecutor(executorService);
 
 		// Config to enable https access to the web-ui
-		boolean enableSSL = config.getBoolean(JobManagerOptions.WEB_SSL_ENABLED) &&	SSLUtils.getSSLEnabled(config);
+		boolean enableSSL = config.getBoolean(WebOptions.SSL_ENABLED) &&	SSLUtils.getSSLEnabled(config);
```
Comment (zentol) [inline flink-runtime-web/src/main/java/org/apache/flink/runtime/webmonitor/WebRuntimeMonitor.java]: whitespace after `&&` seems off.

**Comment by tillrohrmann** [comment]
Thanks for the review @zentol. You're right that we should remove the `key()` method calls.

**Comment by tillrohrmann** [comment]
Merging this PR.

**Pull request closed without merging**